plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,837 @@
+    type: dict
+    suboptions:
+      admin_state:
+        description:
+        - The MCP admin state enables MisCabling Protocol (MCP) on the interface.
+        - The default value is C(enabled).
+        type: str
+        choices: [ enabled, disabled ]
+      strict_mode:
+        description:
+        - The MCP strict mode.
+        - The value is available only when the MCP admin_state is C(enabled).
+        - The default value is C(off).
+        type: str
+        choices: [ 'on', 'off' ]
+        aliases: [ mcp_mode ]
+      initial_delay_time:
+        description:
+        - The MCP initial delay time in seconds.
+        - The default value is 180.
+        - The value must be between 0 and 1800.
+        - The value is available only when the MCP strict_mode is C(on).
+        type: int
+      transmission_frequency_sec:
+        description:
+        - The MCP transmission frequency in seconds.
+        - The default value is 2.
+        - The value must be between 0 and 300.
+        - The value is available only when the MCP strict_mode is C(on).
+        type: int
+      transmission_frequency_msec:
+        description:
+        - The MCP transmission frequency in milliseconds.
+        - The default value is 0.
+        - The value must be between 0 and 999.
+        - The value is available only when the MCP strict_mode is C(on).
+        type: int
+      grace_period_sec:
+        description:
+        - The MCP grace period in seconds.
+        - The default value is 3.
+        - The value must be between 0 and 300.
+        - The value is available only when the MCP strict_mode is C(on).
+        type: int
+      grace_period_msec:
+        description:
+        - The MCP grace period in milliseconds.
+        - The default value is 0.
+        - The value must be between 0 and 999.
+        - The value is available only when the MCP strict_mode is C(on).
+        type: int
+  pfc_admin_state:
+    description:
+    - The Priority Flow Control (PFC) admin state.
+    - The default value is C(auto).
+    type: str
+    choices: [ 'on', 'off', auto ]
+  access_macsec_policy:
+    description:
+    - The access MACsec policy.
+    - The value is available only when the mcp_admin_state is C(enabled).
+    - The MACsec policy must be defined in the same fabric policy template.
+    type: str
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Tenant template.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_macsec_policy
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create an Interface policy group of interface_type physical
```

Comment:
Should we add a full example that also provides some of the list/dict input structures?

Reply:
too many attributes, but added for one of the types